### PR TITLE
Post article comments

### PR DESCRIPTION
**Pull request by Coding-Plumber**

### CODE REVIEW

File: __tests__/articles-article_id.test.js [hunk 1]
```diff
@@ -68,7 +64,7 @@ describe("String input as pathway", () => {
       .get("/api/articles/'22'")
       .then((response) => {
         expect(response.status).toBe(400);
-        expect(response.body.message).toEqual("Invalid input");
+        expect(response.body.message).toEqual("Invalid input, please enter Int");
```
Comment (Augs0) [inline __tests__/articles-article_id.test.js]: This is maybe not the best message for a user. Int means nothing to them

File: __tests__/commentsByArticle.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  });
+
+  describe("GET non existant article_id", () => {
+    it("GET /api/articles/740/comments", () => {
+      return request(app)
+        .get("/api/articles/740/comments")
+        .then((response) => {
+          expect(response.status).toBe(404);
+          expect(response.body).toEqual({ message: "Comments not found" });
```
Comment (Augs0) [inline __tests__/commentsByArticle.test.js]: It might be better to say article not found as this is more reflective of the situation

File: __tests__/commentsByArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          expect(response.body).toBeInstanceOf(Object);
+        });
+    });
+  });
```
Comment (Augs0) [inline __tests__/commentsByArticle.test.js:92]: We also need a test and logic for if we go to an article that exists but has no comments yet (coming back as an empty array)

File: controllers/getArticles.controller.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      }
+    })
+    .catch((error) => {
+      errorHandler(error, req, res);
+    });
+}
+
+async function getCommentsByArticleIdController(req, res) {
+  const { article_id } = req.params;
+  if (isNaN(article_id)) {
```
Comment (Augs0) [inline controllers/getArticles.controller.js:40]: PSQL will throw an error if this is not a number so this if statement isn't needed

File: __tests__/postArticleComment.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+      const response = await request(app)
+        .post(`/api/articles/${articleId}/comments`)
+        .send({ username, body: commentBody });
+
+      expect(response.status).toEqual(201);
+      expect(response.body).toBeInstanceOf(Object);
+      expect(response.body).toEqual({ comment: commentBody });
+
+      const { rows } = await db.query(`SELECT * FROM comments WHERE author = '${username}' AND body = '${commentBody}';`);
```
Comment (Augs0) [inline __tests__/postArticleComment.test.js]: We shouldn't be using a db query in the test file

File: __tests__/postArticleComment.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+afterAll(() => {
+    db.end();
+  });
+  
+  beforeEach(() => {
+    return seed(data);
+  });
+
+  describe('POST /api/articles/:article_id/comments', () => {
+    it('Checks if the user has been created', async () => {
```
Comment (Augs0) [inline __tests__/postArticleComment.test.js]: Why would we be creating a user here?

File: errors/errorHandler.js [hunk 7]
```diff
@@ -1,9 +1,45 @@
...
+  
+    res.status(404).send( { error : err.message } );
+  } else {
+    next(err);
+  }
+}
+
+function postArticleCommentErrors(err, req, res, next) {
+  // Error for inserting a comment to a article_id with a invalid username, eg /api/articles/3/comments
+  if(err.code === '23503')
```
Comment (Augs0) [inline errors/errorHandler.js:30]: This error could happen elsewhere too so try not to be too specific with the error

File: index.js [hunk 8]
```diff
@@ -1,16 +1,30 @@
 const express = require('express');
 const app = express();
+const bodyParser = require('body-parser');
```
Comment (Augs0) [inline index.js]: Not needed in Express anymore. We can use express.json()

**Review by Augs0 — CHANGES_REQUESTED** [review]
Hey Kyle, good work so far. It looks like so much of your other code has come through with this though which made it very hard to read. It might have been better to add all your fixes to main in regards to the files and then make your branch for this task. Your test file looks unfinished as I can't see any happy paths, just tests for if the username doesn't exist, etc. There is also a database call in there that shouldn't be there. Please review the ticket and make sure you have followed the tasks before resubmitting

File: __tests__/postArticleComment.test.js (hunk 5 above)
Comment (Coding-Plumber) [reply __tests__/postArticleComment.test.js]: Took me ages to work out how to do this haha, i wanted to check if the comment has been inserted into the DB and check for it but i suppose we are doing that by checking the res.body aren't we.

File: __tests__/postArticleComment.test.js (hunk 6 above)
Comment (Coding-Plumber) [reply __tests__/postArticleComment.test.js]: I think this was because i misunderstood and i actually made a query to the DB to match the user i didn't realise SQL would do that for us until i got to that part so I've removed the other function but i must have missed this description to remove also, same with the is(NaN) functionality i'll remove those parts

File: __tests__/postArticleComment.test.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+afterAll(() => {
+  db.end();
+});
+
+beforeEach(() => {
+  return seed(data);
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
```
Comment (pippaamy) [inline __tests__/postArticleComment.test.js:16]: only need one describe block for each task!

File: __tests__/postArticleComment.test.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+    expect(response.status).toEqual(201);
+    expect(response.body).toBeInstanceOf(Object);
+    expect(response.body).toEqual({ comment: commentBody });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("Invalid username input", () => {
+    return request(app)
```
Comment (pippaamy) [inline __tests__/postArticleComment.test.js:34]: can we pop the status code expected in the it describe please

File: __tests__/postArticleComment.test.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+afterAll(() => {
+  db.end();
+});
+
+beforeEach(() => {
+  return seed(data);
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("Checks the comment body for the posted comment", async () => {
```
Comment (pippaamy) [inline __tests__/postArticleComment.test.js:17]: can we do another 201 test to test that it ignores unnecessary properties ? :)

File: __tests__/postArticleComment.test.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    expect(response.body).toBeInstanceOf(Object);
+    expect(response.body).toEqual({ comment: commentBody });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("Invalid username input", () => {
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send({ username: "Duncan", body: "BIG DOG" })
```
Comment (pippaamy) [inline __tests__/postArticleComment.test.js:36]: can we add another error test for :
Status 400, invalid ID eg string
Status 404, non existent ID eg 2000
Status 400, missing required field eg no body

**Review by pippaamy — CHANGES_REQUESTED** [review]
Nice work! A few tests to add, could you add them and then pop in another pr please :)
